test/connection.test.ts

```diff
@@ -68,46 +68,222 @@ describe('Connection', function () {
+      this.clientConn.end()
+    })
+
+    it('should complete sending data & money from server on multiple streams when end is called on server side of the connection ', async function() {
       const moneySpy = sinon.spy()
       const dataSpy = sinon.spy()
       this.clientConn.on('stream', (stream: DataAndMoneyStream) => {
-        stream.on('end', () => {
-          console.log('money stream end')
+        stream.setReceiveMax(1000)
+        stream.on('money', (amount) => {
           moneySpy()
+          assert.equal(amount, 200)
+        })
+        stream.on('data', (data: Buffer) => {
+          dataSpy()
+          assert.equal(data.toString(), 'hello')
         })
       })
-      this.clientConn.on('stream', (stream: DataAndMoneyStream) => {
-        stream.on('finish', () => {
-          console.log('data strem end')
+      const serverStreamA = this.serverConn.createStream()
+      const serverStreamB = this.serverConn.createStream()
+      serverStreamA.write('hello')
+      serverStreamA.sendTotal(100)
+      serverStreamB.write('hello')
+      serverStreamB.sendTotal(100)
+
+      await this.serverConn.end()
+      assert.calledTwice(moneySpy)
+      assert.calledTwice(dataSpy)
+    })
+
+    it('should complete sending data & money from client on multiple streams when end is called on client side of the connection ', async function() {
+      const moneySpy = sinon.spy()
+      const dataSpy = sinon.spy()
+      this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.setReceiveMax(1000)
+        stream.on('money', (amount) => {
+          moneySpy()
+          assert.equal(amount, 50)
+        })
+        stream.on('data', (data: Buffer) => {
           dataSpy()
+          assert.equal(data.toString(), 'hello')
         })
       })
-      this.serverConn.createStream().setSendMax(100)
-      this.serverConn.createStream().write('hello')
+      const clientStreamA = this.clientConn.createStream()
+      const clientStreamB = this.clientConn.createStream()
+      clientStreamA.write('hello')
+      clientStreamA.sendTotal(100)
+      clientStreamB.write('hello')
+      clientStreamB.sendTotal(100)
 
-      console.log('about to end')
       await this.clientConn.end()
-      console.log('ended')
+      assert.calledTwice(moneySpy)
+      assert.calledTwice(dataSpy)
+    })
+
+    it('should error on sending data from client after the server end is called', async function() {
+      const clientStreamError = sinon.spy()
+      const serverStreamData = sinon.spy()
+      this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.setReceiveMax(1000)
+        stream.on('data', serverStreamData)
+      })
+      const clientStream = this.clientConn.createStream()
+      clientStream.on('error', clientStreamError)
+      clientStream.write('hello')
+      await this.serverConn.end()
+      clientStream.write('hello')
+
+      assert.calledOnce(clientStreamError)
```

Comment:
You should do `assert.calledWithMatch` to make sure the error message is what you expect